src/components/user/unavailability/unavailability.service.ts

```diff
@@ -165,14 +165,37 @@ export class UnavailabilityService {
     session: Session
   ): Promise<Unavailability> {
     const unavailability = await this.readOne(input.id, session);
+    const query = this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .match([
+        node('user'),
+        relation('out', '', 'unavailability', { active: true }),
+        node('node', { id: input.id }),
+      ])
+      .return('user.id as unavailabilityUserId')
+      .asResult<{
+        unavailabilityUserId: string;
+      }>();
+    const result = await query.first();
 
-    return await this.db.sgUpdateProperties({
-      session,
-      object: unavailability,
-      props: ['description', 'start', 'end'],
-      changes: input,
-      nodevar: 'unavailability',
-    });
+    if (result?.unavailabilityUserId === session.userId) {
+      return await this.db.updateProperties({
+        type: 'Unavailability',
+        object: unavailability,
+        props: ['description', 'start', 'end'],
+        changes: input,
+        skipAuth: true,
+      });
+    } else {
+      return await this.db.sgUpdateProperties({
+        session,
+        object: unavailability,
+        props: ['description', 'start', 'end'],
+        changes: input,
+        nodevar: 'unavailability',
+      });
+    }
```

Comment:
This, and the other places like this, can be simplified.

```suggestion
    return await this.db.updateProperties({
      type: Unavailability,
      object: unavailability,
      props: ['description', 'start', 'end'],
      changes: input,
      // assume current user has permission to edit themselves since the auth doesn't account for this yet.
      skipAuth: result?.unavailabilityUserId === session.userId,
    });
```